### PR TITLE
Move thumb nailing out of -go so that we can deal with HEIF in native layer

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: repo/wallet/data.go [hunk 1]
```diff
@@ -84,25 +85,27 @@ func PinPhoto(reader io.Reader, fname string, nd *core.IpfsNode, apiHost string)
...
+	t, err := ioutil.ReadAll(thumb)
+	if err != nil {
+		return nil, err
+	}
+
+	tb := bytes.NewReader(t)
+	tb.Seek(0, 0)
+
+	// add the thumbnail
+	addFileToDirectory(dirb, tb, "thumb.jpg", nd)
```
Comment (sanderpick) [inline repo/wallet/data.go]: hey dude, nice! don't think you need to create a new reader here, just use thumb direct in addFileToDirectory

File: repo/wallet/data.go [hunk 2]
```diff
@@ -4,14 +4,14 @@ import (
 	"io"
 	"time"
 	"bytes"
-	"image/jpeg"
+	_ "image/jpeg"
 	_ "image/png" // register other possible image types for decoding
 	_ "image/gif"
 	"encoding/json"
-	"image"
+	_ "image"
```
Comment (sanderpick) [inline repo/wallet/data.go]: apparently it's bad practice to use the underscore method import unless you are just "loading codecs" - if you dont need these just delete em

**Comment by asutula** [comment]
Ok, so I have the RN layer creating a JPEG thumb. The thumb is written to a path that can be passed into the Go layer. So the Go pin method should now take two path, one for the thumb and one for the full image.

**Comment by asutula** [comment]
Ah cool, looks like that's the way the code is currently working. @andrewxhill you want to address @sanderpick's comments then we'll get this merged? Can't wait to try it on a real device!

File: repo/wallet/data.go (hunk 1 above)
Comment (andrewxhill) [reply repo/wallet/data.go]: okay, i guess we didn't need the orig reader copy either

File: repo/wallet/data.go (hunk 2 above)
Comment (andrewxhill) [reply repo/wallet/data.go]: done

**Comment by andrewxhill** [comment]
okay, should be gtg

**Comment by asutula** [comment]
Do you know how to generate the updated `Mobile.framework`?

**Pull request merged**